inst/scripts/importPDB.R

```diff
@@ -0,0 +1,158 @@
+                        tkdestroy(win1)
+                    })
+
+butSave <- tkbutton(win1, text = "Pause & Save",
+                    command = function() {
+                        assign("stoploop", TRUE, envir = .GlobalEnv)
+                        print("Finishing the current gene..")
+                        saveRDS(genes, file = "inst/tmp/genes.rds")
+                        saveRDS(GeneData, "inst/extdata/GeneData.rds")
+                        tkdestroy(win1)
+                    })
+tkgrid(butStop)
+tkgrid(butSave)
+stoploop <- FALSE
+
+readRDS(MT.rds)
+
+#=============================(LOAD & CONTINUE)/START==========================
+# Check if there was a saved progress from before.
+if (file.exists("inst/tmp/genes.rds")){
+    genes <- readRDS("inst/tmp/genes.rds")
+
+} else {
+    genes <- list(gene names from MT)
+}
+
+while (length(genes > 0) && !stoploop){
+    results <- list()
+
+    #=========================GET TARGET SEQUENCE==============================
+
+    # The current gene is always the first element of the list.
+    # This element will be removed from the list of genes at the end of each
+    # iteration of the while loop.This is necessary for pausing the process and
+    # continuing it later.
+
+    curr_gene <- genes[[1]]
+
+    # Translate the cDNA of the gene to its aa sequence
+    targetAA <- AAString(paste(a.a. sequence of the gene, collapse = "")
+
+    for (pdbID_chainID matched to gene){
+        pdbID <- pdbid retrieved from the MT object
+        chainID <- chainid retrieved from the MT object
+
+        #=====================READ SUBJECT PDB FILE============================
+
+        # download the PDB file.
+        subject <- read.pdb(pdbID)
+        # extract the desired chain and get its aa sequence.
+        chainAAIndices <- atom.select(subject,
+                                      "calpha",
+                                      chain = chainID,
+                                      value = TRUE)
+        subjectAA <- AAString(paste(aa321(chainAAIndices$atom$resid), collapse = ""))
+
+        #=====================ALIGNMENT========================================
+
+        # retrieve the start and end indices for the target & subject sequences:
+        start_target <- the starting index of the alignment in targetAA from MT
+        end_target <- the last index of the alignment in targetAA from MT
+        start_subject <- the starting index of the alignment subjectAA from MT
+        end_subject <- the last index of the alignment in target AA from MT
+
+        # perform pairwise alignment with default parameters.
+        ali <- pairwiseAlignment(subjectAA[start_subject, end_subject],
+                                 targetAA[start_target, end_target])
+
+        #======================================================================
+
+        # create a data frame with nmatch(ali) rows.
+        DF <- data.frame("ResID" = c(1:nmatch(ali)),

```

Comment:
done.
But would you think the other way that I mentioned might be better and possible faster? 

> But I think I just figured that I can maybe change the column names at the end, when all the for loops are done computing. I will have to try this and see if it works.

But I think it might look a bit weird/confusing in the code, since I will be naming the columns twice: once when I create the empty data frame, and once after the data frame has been populated.